src/test/java/com/g4s8/ghman/data/PgUserITCase.java

```diff
@@ -16,32 +16,123 @@
- *  methods of {@link com.g4s8.ghman.data.PgUser} class. Do not forget about
- *  all possible exceptions.
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
  */
 @ExtendWith(DatabaseExtension.class)
 final class PgUserITCase {
 
     @Test
     void throwsExceptionIfUserIsNotFound() {
         new Assertion<>(
-            "Unexpected exception while selecting not existing user",
-            () -> new PgUser(DatabaseExtension.dataSource(), 0).github(),
+            "Should throw exception while selecting not existing user",
+            () -> new PgUser(DatabaseExtension.dataSource(), -1).github(),
             new Throws<>(
                 "Failed to select token",
                 IOException.class
             )
         ).affirm();
     }
 
+    @Test
+    void returnsUserId() {
+        final long uid = 45;
+        new Assertion<>(
+            "Should return user id",
+            new PgUser(DatabaseExtension.dataSource(), uid).uid(),
+            new IsEqual<>(uid)
+        ).affirm();
+    }
+
+    @Test
+    void throwsExceptionIfTokenIsEmpty() throws IOException {
+        final User user = new PgUsers(DatabaseExtension.dataSource())
+            .user(getChat(458L, "test1"));
+        user.authorize("");
+        new Assertion<>(
+            "Should throw exception if token is empty",
+            user::github,
+            new Throws<>(
+                "Github token was not found",
+                GhAuthException.class
+            )
+        ).affirm();
+    }
+
+    @Test
+    void returnsGhUser() throws IOException {
+        final User user = new PgUsers(DatabaseExtension.dataSource())
+            .user(getChat(408L, "test2"));
+        final String token = "some_token";
+        user.authorize(token);
+        new Assertion<>(
+            new FormattedText(
+                "Should return github API instance with token %s", token
+            ).toString(),
+            user.github().github(),
+            new IsEqual<>(new RtGithub(token))
+        ).affirm();
+    }
+
+    @Test
+    void returnsTelegramId() throws IOException {
+        final long tid = 409L;
+        final User user = new PgUsers(DatabaseExtension.dataSource())
+            .user(getChat(tid, "test3"));
+        new Assertion<>(
+            "Should return telegram id",
+            user.tid(),
+            new IsEqual<>(String.valueOf(tid))
+        ).affirm();
+    }
+
+    @Test
+    void returnsTelegram() throws IOException {
+        final long tid = 401L;
+        final String uname = "test4";
+        final User user = new PgUsers(DatabaseExtension.dataSource())
+            .user(getChat(tid, uname));
+        new Assertion<>(
+            "Should return json with telegram name and id",
+            user.telegram(),
+            new AllOf<>(
+                new IterableOf<>(
+                    new JsonHas("firstName", new JsonValueIs(uname)),
+                    new JsonHas("uid", new JsonValueIs(tid))
```

Comment:
@olenagerasimova isn't there 2 other field in this json datastructure?

Reply:
@victornoel no, my mocked chat has only first name and user id

Reply:
@olenagerasimova ah, I understand, yep it makes sense